helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -92,11 +92,14 @@ protected AssignmentMetadataStore(BucketDataAccessor bucketDataAccessor, String
     return _bestPossibleAssignment;
   }
 
-  public void persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
+  /**
+   * @return true if a new baseline was persisted.
+   */
+  public boolean persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
     // TODO: Make the write async?
     // If baseline hasn't changed, skip writing to metadata store
     if (compareAssignments(_globalBaseline, globalBaseline)) {
-      return;
+      return false;
```

Comment:
This behavior might be confusing. This is not ideal... usually people would think if I get a false, that means that a write to metadata store failed (for example, ZK write failure). But here, that's not necessarily the case. In this case, we decide not to write it on purpose because there's no diff.

How should we differentiate this behavior clearly? No diff != Write failure and should make each case be expressed distinctly?

Reply:
Write failure -> exception
Write skipped -> return false
How about this?

Reply:
We have such logic in ZkClient.updater()

Reply:
I just removed my previous comment about ZK updater. That was not right.
What @pkuwm mentioned in the other comment can help with it. But moving this check outside is too verbose and complicated the store class design.
Let me know if you think the following behavior is acceptable or not:
Write failure -> throws exception
Write skipped -> return false

Reply:
Added a TODO here. Also modified comments.